pkg/controller/instance/instance_controller.go

```diff
@@ -186,30 +174,21 @@ func (r *Reconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 		return reconcile.Result{}, err // OV not found has to be retried because it can really have been created after Instance
 	}
 
-	// ---------- 2. Check if we should start execution of new plan ----------
+	// ---------- 2. Execute scheduled plan if exists ----------
+	ensurePlanStatusInitialized(instance, ov)
```

Comment:
The function docs hint that this should only get called for a fresh instance or after an OV was updated. As a result it was guarded by `if i.NoPlanEverExecuted() || isUpgradePlan(planName)` in `startPlanExecution`. Calling this function will update plan/phase/step statuses with current timestamps. Is this intended here?

Reply:
Good catch, I didn't see that the `lastUpdatedTimestamp` was being modified. Ideally, this shouldn't happen, and the method should be idempotent.